sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -656,6 +661,28 @@
       return toBuilder().setCommitOffsetsInFinalizeEnabled(true).build();
     }
 
+    /**
+     * Set additional parameters for KafkaUnboundedReader.offsetConsumer, added by BEAM-6285.
```

Comment:
nit. It is better to write JavaDoc that explains functionality to the end user who should be assumed to not know the code at all. It is better not to refer to specific bug reports since they may think the description here may be incomplete. 

E.g. here the first line is pretty opaque to most users.  It could start with something like: 
   
     Set additional configuration for secondary 'offset consumer' used inside the reader. Each KafkaIO reader has two 
     consumers,  ... 

Remaining JavaDoc looks fairly good, But please avoid referring internal variables like KafkaUnboundedReader.offsetConsumer .